bgpd/bgp_route.c

```diff
@@ -10014,19 +10008,49 @@ DEFUN (show_ip_bgp,
 		return bgp_show_route_map(vty, bgp, argv[idx + 1]->arg, afi,
 					  safi, bgp_show_type_route_map);
 
+	/* prefix-longer */
+	if (argv_find(argv, argc, "A.B.C.D/M", &idx)
+	    || argv_find(argv, argc, "X:X::X:X/M", &idx))
+		return bgp_show_prefix_longer(vty, bgp, argv[idx]->arg, afi,
+					      safi,
+					      bgp_show_type_prefix_longer);
+
+	return CMD_WARNING;
+}
+
+/* Handle show ip bgp community-list separately due to
+ * show bgp community-list <(1-500)|WORD> introduced in
+ * https://github.com/FRRouting/frr/issues/3308
+ */
+DEFUN(show_ip_bgp_community_list, show_ip_bgp_community_list_cmd,
+      "show ip bgp [<view|vrf> VIEWVRFNAME] [" BGP_AFI_CMD_STR
```

Comment:
ip should be [ip]

Reply:
This is why I moved only `with ip` definition to avoid conflicts with `show [ip] bgp ...`. You can see the details in https://github.com/FRRouting/frr/issues/3308#issuecomment-438046994

Reply:
I suggest following  DEFUN (show_ip_bgp_large_community_list, 
as a pattern.

Reply:
Actually that code which is under `show_ip_bgp_large_community_list` is parsed the same way as with `bgp_vty_find_and_parse_afi_safi_bgp()`. Would you mind to tell me which part of the code you suggest to follow?